stakingreward/stakingreward.go

```diff
@@ -30,56 +28,38 @@ type calculator struct {
 	TicketReward float64
 	RewardPeriod float64
 
-	ChainParams      *chaincfg.Params
-	Version          string
-	NetName          string
 	MeanVotingBlocks int64
 
-	dcrdChainSvr *rpcclient.Client
-	reorgLock    sync.Mutex
+	reorgLock sync.Mutex
 }
 
-func New(dcrdClient *rpcclient.Client, webServer *web.Server, viewFolder string,
-	xcBot *exchanges.ExchangeBot, params *chaincfg.Params) (*calculator, error) {
-
-	if viewFolder == "" {
-		viewFolder = "./pkgs/stakingreward/views"
-	}
-
-	exp := &calculator{
-		templates:    web.NewTemplates(viewFolder, false, []string{"extras"}, web.MakeTemplateFuncMap(params)),
-		webServer:    webServer,
-		xcBot:        xcBot,
-		ChainParams:  params,
-		dcrdChainSvr: dcrdClient,
-		stakePerc:    45.0,
+func New(client *dcrd.Dcrd, webServer *web.Server, xcBot *exchanges.ExchangeBot) (*calculator, error) {
+	calc := &calculator{
+		webServer: webServer,
+		xcBot:     xcBot,
+		client:    client,
+		stakePerc: 45.0, // why this?
```

Comment:
This should be 55.38, representing the amount of DCR that is staked at the moment.
But it can be removed as `ConnectBlock` is called immediately which will override this assignment.